### ISSUE TITLE
NPE: ColumnIndices.getColumnInfo(java.lang.Class) after upgrading to 0.84.0

### ISSUE THREAD

**Issue by RobertoArtiles** [realm/realm-java#1646]
Hi team,

After upgrading from 0.82.2 to 0.84.0 got this exception during the first app start:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'io.realm.internal.ColumnInfo io.realm.internal.ColumnIndices.getColumnInfo(java.lang.Class)' on a null object reference
```

Didn't happen during the next starts.
Same happened when I deployed the app on another device with the same 0.82.2 realm version.

Here is the piece of code which triggered the crash on the first device (called on the worker thread):
```java
realm.where(PinRealm.class)
                    .equalTo("token", pinToken)
                    .findFirst();
```

And here is the piece of code which triggered the crash on another device (called on the main thread):
```java
realm.where(PinRealm.class)
                    .equalTo("deleted", false)
                    .equalTo("inboxState", NEW.ordinal())
                    .findAll();
```

**Comment by zaki50** [comment]
Hi @RobertoArtiles I'll try to reproduce that.
Is it possible for you to make a simple project to reproduce this problem?

**Comment by zaki50** [comment]
@RobertoArtiles One more question. How do you get `Realm` instance?

**Comment by RobertoArtiles** [comment]
>Hi @RobertoArtiles I'll try to reproduce that.
Is it possible for you to make a simple project to reproduce this problem?

I can try.

>@RobertoArtiles One more question. How do you get Realm instance?

First in ```Application.onCreate()``` I set the default config like this:

```java
        RealmConfiguration config = new RealmConfiguration.Builder(this)
                .deleteRealmIfMigrationNeeded()
                .schemaVersion(RealmConfig.REALM_SCHEME_VERSION)
                .build();
        Realm.setDefaultConfiguration(config);
```

Then whenever I need a Realm instance I use ```Realm.getDefaultInstance()```.

**Comment by zaki50** [comment]
@RobertoArtiles thanks!

**Comment by diegomontoya** [comment]
Same NPE. 

```
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'io.realm.internal.ColumnInfo io.realm.internal.ColumnIndices.getColumnInfo(java.lang.Class)' on a null object reference
       at io.realm.Realm.getColumnInfo(Realm.java:727)
       at io.realm.RealmMyClassRealmProxy.copyOrUpdate(RealmMyClassRealmProxy.java:1671)
       at io.realm.RealmModuleMyClassMediator.copyOrUpdate(RealmModuleMyClassMediator.java:97)
       at io.realm.Realm.copyOrUpdate(Realm.java:1130)
       at io.realm.Realm.copyToRealmOrUpdate(Realm.java:763)
       at com.fictionpress.fanfiction.realm.table.RealmTableMyClass$1$1.execute(RealmTableMyClass.java:110)
       at io.realm.Realm.executeTransaction(Realm.java:1034)
       at com.fictionpress.fanfiction.realm.table.RealmTableMyClass$1.run(RealmTableMyClass.java:107)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
```

Using ---> to note crash line.

```java
    public static RealmMyClass copyOrUpdate(Realm realm, RealmMyClass object, boolean update, Map<RealmObject,RealmObjectProxy> cache) {
        if (object.realm != null && object.realm.getPath().equals(realm.getPath())) {
            return object;
        }
        RealmMyClass realmObject = null;
        boolean canUpdate = update;
        if (canUpdate) {
            Table table = realm.getTable(RealmMyClass.class);
            long pkColumnIndex = table.getPrimaryKey();
            long rowIndex = table.findFirstLong(pkColumnIndex, object.getStoryId());
            if (rowIndex != TableOrView.NO_MATCH) {
                ------> realmObject = new RealmMyClassRealmProxy(realm.getColumnInfo(RealmMyClass.class));
                realmObject.realm = realm;
                realmObject.row = table.getUncheckedRow(rowIndex);
                cache.put(object, (RealmObjectProxy) realmObject);
            } else {
                canUpdate = false;
            }
        }
```

**Comment by RobertoArtiles** [comment]
In my case the full stacktrace is this:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'io.realm.internal.ColumnInfo io.realm.internal.ColumnIndices.getColumnInfo(java.lang.Class)' on a null object reference
       at io.realm.RealmQuery.<init>(RealmQuery.java:92)
       at io.realm.Realm.where(Realm.java:828)
```

**Comment by zaki50** [comment]
I found a possible scenario of this NPE.

@RobertoArtiles Does your code have another thread which uses Realm instance?

**Comment by RobertoArtiles** [comment]
@zaki50 Plenty :)

**Comment by zaki50** [comment]
I made a PR #1648 to fix this issue.
After that PR merged into master, please confirm that this PR fixes the issue by using Snapshot build.

https://github.com/realm/realm-java/blob/master/README.md#using-snapshots

Snapshot build is automatically deployed once a day, so you must wait the next deploy. Sorry for that.
To make sure that gradle uses latest snapshot build, please ass `--refresh-dependencies` option to gradle.

    ./gradlew assemble --refresh-dependencies

**Comment by zaki50** [comment]
fix is merged into master.

**Comment by zaki50** [comment]
I close this issue.

If NPE happens again, feel free to reopen this.

**Comment by RobertoArtiles** [comment]
@zaki50 Thanks. Seems like the bug is fixed.

**Comment by zaki50** [comment]
@RobertoArtiles :+1:

**Comment by zaki50** [comment]
We've just released Realm Java 0.84.1 which includes fix of this issue.
